### PR TITLE
[tt-train] Expand ttml.init with in-place and advanced initializers

### PR DESCRIPTION
**Pull request by imichalakTT**
### Problem description

`ttml.init` only had four basic factory functions (`uniform`, `normal`, `zeros`, `ones`). No in-place variants, no xavier/kaiming, no `constant`.

### What's changed

- Add in-place variants (`uniform_`, `normal_`, `constant_`, `zeros_`, `ones_`, `xavier_uniform_`, `xavier_normal_`, `kaiming_uniform_`, `kaiming_normal_`) mirroring `torch.nn.init`
- Add factory variants for `constant`, `xavier_uniform`, `xavier_normal`, `kaiming_uniform`, `kaiming_normal`
- Rename `low`/`high` to `a`/`b` across `rand_op` and nanobind to match PyTorch
- Make `a`/`b`/`mean`/`std` positional in `ttml.ops.rand`/`randn` bindings (`seed`/`dtype`/`layout` stay keyword-only)
- Add Parameter Initialization section to `TTML_ONBOARDING.md` with Python and C++ examples

### Checklist

- tt-train C++ and Python tests (https://github.com/tenstorrent/tt-metal/actions/runs/23482539755)

### CODE REVIEW

**Review by dyurshevichTT — APPROVED** [review]
LGTM

File: tt-train/sources/ttml/nanobind/nb_ops.cpp [hunk 1]
```diff
@@ -369,9 +369,9 @@ void py_module(nb::module_& m) {
         "rand",
         &ttml::ops::rand,
         nb::arg("shape"),
+        nb::arg("a") = 0.0f,
+        nb::arg("b") = 1.0f,
         nb::kw_only(),
```
[the remark this answers is not in the document]
Comment (imichalakTT) [reply tt-train/sources/ttml/nanobind/nb_ops.cpp:374]: `ttml.ops.rand` is relatively new for this to not be an issue, better to align now

File: tt-train/sources/ttml/ttml/init.py [hunk 2]
```diff
@@ -73,12 +202,201 @@ def init_fn(shape):
...
+# ---------------------------------------------------------------------------
+# In-place variants (fill existing tensor, return it)
+# ---------------------------------------------------------------------------
+
+
+def uniform_(tensor, a: float = 0.0, b: float = 1.0):
+    """Fill tensor in-place with values from uniform distribution [a, b)."""
+    t = uniform(a, b)(tensor.shape())
+    tensor.set_value(t.get_value())
+    return tensor
```
[the remark this answers is not in the document]
Comment (imichalakTT) [reply tt-train/sources/ttml/ttml/init.py:323]: solved in https://github.com/tenstorrent/tt-metal/pull/40507/changes/459085e87628ed345b279a3023a330df4d0917ab

**Review by philei-tt — APPROVED** [review]
Maybe add a note somewhere that in-place is not really inplace (it will make another allocation) for now
